client/matching/client.go

```diff
@@ -232,6 +241,49 @@ func (c *clientImpl) ListTaskListPartitions(
 	return client.ListTaskListPartitions(ctx, request, opts...)
 }
 
+func (c *clientImpl) GetTaskListsByDomain(
+	ctx context.Context,
+	request *types.GetTaskListsByDomainRequest,
+	opts ...yarpc.CallOption,
+) (*types.GetTaskListsByDomainResponse, error) {
+	opts = common.AggregateYarpcOptions(ctx, opts...)
+	clients, err := c.clientIterator()
+	if err != nil {
+		return nil, err
+	}
+
+	var futures []future.Future
+	for _, client := range clients {
+		future, settable := future.NewFuture()
+		go func() {
+			settable.Set(client.(Client).GetTaskListsByDomain(ctx, request, opts...))
+		}()
+
+		futures = append(futures, future)
+	}
+
+	taskListMap := make(map[string]struct{})
+	for _, future := range futures {
+		var resp *types.GetTaskListsByDomainResponse
+		if err = future.Get(ctx, &resp); err != nil {
+			return nil, err
+		} else {
+			for _, tl := range resp.TaskListNames {
+				if _, ok := taskListMap[tl]; !ok {
+					taskListMap[tl] = struct{}{}
+				}
+			}
+		}
```

Comment:
nit: remove else and outdent the for loop?

Reply:
updated